spec/syntax.md

```diff
@@ -843,15 +843,62 @@ option     = identifier o "=" o (literal / variable)
+                  / %x2010-2027   ;          omit BidiControl %x200E-200F
+                  / %x2030-205E   ;          omit Whitespace %x2028-2029 %x202F, BidiControl %x202A-202E
+                  / %x2060-2065   ;          omit Whitespace %x205F
+                  / %x206A-2FFF   ;          omit BidiControl %x2066-2069
+                  / %x3001-D7FF   ;          omit Whitespace %x3000
+                  / %xF900-FDCF   ;          omit Cs %xD800-DFFF, Co %xE000-F8FF
+                  / %xFDF0-FFFD   ;          omit NChar %xFDD0-FDEF
+                  / %x10000-1FFFD ;          omit NChar %xFFFE-FFFF
+                  / %x20000-2FFFD ;          omit NChar %x1FFFE-1FFFF
+                  / %x30000-3FFFD ;          omit NChar %x2FFFE-2FFFF
+                  / %x40000-4FFFD ;          omit NChar %x3FFFE-3FFFF
+                  / %x50000-5FFFD ;          omit NChar %x4FFFE-4FFFF
+                  / %x60000-6FFFD ;          omit NChar %x5FFFE-5FFFF
+                  / %x70000-7FFFD ;          omit NChar %x6FFFE-6FFFF
+                  / %x80000-8FFFD ;          omit NChar %x7FFFE-7FFFF
+                  / %x90000-9FFFD ;          omit NChar %x8FFFE-8FFFF
+                  / %xA0000-AFFFD ;          omit NChar %x9FFFE-9FFFF
+                  / %xB0000-BFFFD ;          omit NChar %xAFFFE-AFFFF
+                  / %xC0000-CFFFD ;          omit NChar %xBFFFE-BFFFF
+                  / %xD0000-DFFFD ;          omit NChar %xCFFFE-CFFFF
+                  / %xE0000-EFFFD ;          omit NChar %xDFFFE-DFFFF,
+                                  ;          omit NChar %xEFFFE-EFFFF %xFFFFE-FFFFF %x10FFFE-10FFFF,
+                                  ;          omit Co %xF0000-FFFFD %x100000-10FFFD
+name-char  = name-start / DIGIT
+                  / %x2D-2E       ; 【-.】    omit Cc %x0-1F, Whitespace 【 】, Ascii 【!"#$%&'()*+,】
 ```
 
+> [!NOTE]
+> Syntactically, the definitions of `identifier` and `name-char` provide backwards compatibility over time by allowing a stable,
+> wide range of characters.
+> So when there is a new character in a version of Unicode, it can be used in any conformant implementation of Message Format.
+> The definition currently excludes:
+> * Most ASCII except for letters and characters used for numbers
+>    * This avoids conflicts with syntax characters, and reserves some characters for future syntax.
+> * Bidirectional controls (`Bidi_C`)
+> * Control characters (`GC=Cc`, but not Format characters: `GC=Cf`)
+> * Whitespace characters (`WSpace`)
+> * Isolated Surrogate characters (`GC=Cs`)
+> * Private use characters (`GC=Co`)
+> * Non-Characters (`NChar`)
+>
+> Although syntactically a wide range of characters are included,
+> when function and implementations and message authors are creating new identifiers (for functions, options, variables, …),
+> it is strongly recommended that they conform to the following to minimize confusion.
+> These are also recommended for Message Format linter implementations.
```

Comment:
This is normative language in a "NOTE", which is verboten in our specification. There is no need to make this a note, since this is the specification and is meant to describe what is going on. Something like:

```suggestion
This syntax allows a wide range of characters in _names_ and _identifiers_.
Implementers and authors of _functions_ and _messages_,
including _functions_, _options_, and _operands_ (variable names),
SHOULD avoid creating _names_ that could produce confusion or harm usability
by choosing names consistent with the following guidelines.
MessageFormat tools, such as linters, SHOULD warn when _names_ chosen by users
violate these constraints.
```

(Note that it is `MessageFormat` not `Message Format`)